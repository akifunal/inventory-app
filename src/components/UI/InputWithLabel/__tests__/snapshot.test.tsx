import { render } from '@testing-library/react';
import InputWithLabel from '../index';

describe('InputWithLabel component', () => {
	it('should render and match the snapshot', () => {
		// Arrange and Act
		const { asFragment } = render(
			<InputWithLabel id='test-first' label='test first' value='test' />
		);

		// Assert
		expect(asFragment()).toMatchSnapshot();
	});

	it('should render and match the snapshot when success', () => {
		// Arrange and Act
		const { asFragment } = render(
			<InputWithLabel
				id='test-success'
				label='test success'
				status='success'
				value='test2'
			/>
		);

		// Assert
		expect(asFragment()).toMatchSnapshot();
	});

	it('should render and match the snapshot when error', () => {
		// Arrange and Act
		const { asFragment } = render(
			<InputWithLabel
				id='test-error'
				label='test error'
				status='error'
				value='test3'
			/>
		);

		// Assert
		expect(asFragment()).toMatchSnapshot();
	});
});
