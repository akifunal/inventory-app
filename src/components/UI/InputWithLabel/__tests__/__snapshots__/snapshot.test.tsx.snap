// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithLabel component should render and match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <label
      class="mb-2 block text-sm font-medium text-gray-700 dark:text-gray-500"
      for="test-first"
    >
      test first
    </label>
    <input
      class="block w-full rounded-lg border border-gray-500 bg-gray-50 p-2.5 text-sm text-gray-900 placeholder-gray-700 focus:border-gray-500 focus:ring-gray-500 dark:border-gray-500 dark:bg-gray-700 dark:text-gray-400 dark:placeholder-gray-500"
      id="test-first"
      type="text"
      value="test"
    />
  </div>
</DocumentFragment>
`;

exports[`InputWithLabel component should render and match the snapshot when error 1`] = `
<DocumentFragment>
  <div>
    <label
      class="mb-2 block text-sm font-medium text-red-700 dark:text-red-500"
      for="test-error"
    >
      test error
    </label>
    <input
      class="block w-full rounded-lg border border-gray-500 bg-gray-50 p-2.5 text-sm text-gray-900 placeholder-gray-700 focus:border-gray-500 focus:ring-gray-500 dark:border-gray-500 dark:bg-gray-700 dark:text-gray-400 dark:placeholder-gray-500"
      id="test-error"
      type="text"
      value="test3"
    />
  </div>
</DocumentFragment>
`;

exports[`InputWithLabel component should render and match the snapshot when success 1`] = `
<DocumentFragment>
  <div>
    <label
      class="mb-2 block text-sm font-medium text-green-700 dark:text-green-500"
      for="test-success"
    >
      test success
    </label>
    <input
      class="block w-full rounded-lg border border-gray-500 bg-gray-50 p-2.5 text-sm text-gray-900 placeholder-gray-700 focus:border-gray-500 focus:ring-gray-500 dark:border-gray-500 dark:bg-gray-700 dark:text-gray-400 dark:placeholder-gray-500"
      id="test-success"
      type="text"
      value="test2"
    />
  </div>
</DocumentFragment>
`;
